packages/interactivity/src/vdom.ts

```diff
@@ -47,6 +47,8 @@ const directiveParser = new RegExp(
 const nsPathRegExp = /^([\w_\/-]+)::(.+)$/;
 
 export const hydratedIslands = new WeakSet();
+const nodesToRemove = new Set< Node >();
+const nodesToReplace = new Set< Node >();
```

Comment:
I think we have to move these inside `toVdom()`. 🤔 

The `toVdom()` function is called recursively for `template` tags, so it could empty those sets too soon and leave nodes in the `template`'s parent tree untouched.

Also, that way, clearing these sets at the end of the `toVdom()` function is not required. The garbage collector can take care of deleting them.

Reply:
Yeah, I had them there at first, but I don't know why I moved them out. Good catch.

Fixed in https://github.com/woocommerce/gutenberg/pull/7/commits/497bcf61aae22977a3ec4d79a2468c6c3e20990a.